packages/core/src/odata-common/entity-builder.ts

```diff
@@ -49,24 +63,44 @@ export class EntityBuilder<EntityT extends Entity, JsonT> {
    * @param json - Representation of the entity in JSON format.
    * @returns Entity constructed from JSON representation.
    */
-  public fromJson(json: Partial<JsonT>): EntityT {
+  public fromJson(json: FromJsonType<JsonT>): EntityT {
     const entityBuilder = this._entityConstructor.builder();
-    Object.entries(json).forEach(([key, value]) => {
-      if (typeof entityBuilder[key] === 'function') {
-        entityBuilder[key](value);
-      }
-      if (key === '_customFields') {
-        entityBuilder.withCustomFields(value as Record<string, any>);
-      }
-    });
+    const entityConstructor = this._entityConstructor;
+
+    const [entityEntries, customEntries] = partition(
+      Object.entries(json),
+      ([key]) => typeof entityBuilder[key] === 'function'
+    );
+
+    entityEntries.forEach(([key, value]) => {
+      const propertyValue = isNavigationProperty(key, entityConstructor)
+        ? buildNavigationPropertyFromJson(key, value, entityConstructor)
+        : value;
+
+      entityBuilder[key](propertyValue);
+    }, entityBuilder);
```

Comment:
question:
Can I remove the second parameter `entityBuilder` of the `forEach` in line 81?